app/Core/HtmxController.php

```diff
@@ -69,11 +71,11 @@ private function executeActions(): void
 
         $action = Str::camel($this->incomingRequest->query->get('id', 'run'));
 
-        if (! method_exists($this, $action)) {
-            throw new Error("Method $action doesn't exist.");
+        if (! method_exists($this, $action) && ! method_exists($this, 'run')) {
+            throw new Error("Method $action doesn't exist and no fallback method.");
         }
 
-        $fragment = $this->$action();
+        $fragment = method_exists($this, $action) ? $this->$action() : $this->run();
```

Comment:
We'll also need to check that the run method exists. Had this issue a few times.